go/libraries/doltcore/sqle/dsess/transactions.go

```diff
@@ -275,6 +258,38 @@ func (tx *DoltTransaction) doCommit(
 	return nil, nil, datas.ErrOptimisticLockFailed
 }
 
+func handleConflictSystemVariables(ctx *sql.Context, workingSet *doltdb.WorkingSet) error {
+	forceTransactionCommit, err := ctx.GetSessionVariable(ctx, ForceTransactionCommit)
+	if err != nil {
+		return err
+	}
+
+	allowCommitConflicts, err := ctx.GetSessionVariable(ctx, AllowCommitConflicts)
+	if err != nil {
+		return err
+	}
+
+	if !(allowCommitConflicts.(int8) == 1 || forceTransactionCommit.(int8) == 1) {
+		workingRoot := workingSet.WorkingRoot()
+		hasConflicts, err := workingRoot.HasConflicts(ctx)
+		if err != nil {
+			return err
+		}
+		if hasConflicts {
+			return doltdb.ErrUnresolvedConflicts
+		}
+		hasConstraintViolations, err := workingRoot.HasConstraintViolations(ctx)
```

Comment:
This should actually be covered by forceTransactionCommit, but not by allowCommitConflicts

Reply:
Also add a TODO here: you should be able to commit even with constraint violations depending on the type by setting various session vars.

For example, if you have `foreign_key_checks=0`, you should be able to commit foreign key violations, but not other kinds of constraint violations. We'll add other vars to control enforcing other kinds of checks.

And of course the `force` var should override all checks and commit no matter what